.github/workflows/deployment.yaml

```diff
@@ -108,11 +110,13 @@ jobs:
       # Setup gcloud CLI
       - name: Setup gcloud CLI
         uses: google-github-actions/setup-gcloud@v0
+        with:
+          install_components: kubectl
+  
       - name: Setup Kubernetes config
         run: |
-          gcloud components install kubectl
-          kubectl config set-context current --namespace=argocd
-          kubectl config use-context current
+          gcloud container clusters get-credentials "${{ inputs.gke_cluster }}" --zone "${{ inputs.gke_zone }}"
+          kubectl config set-context --current --namespace=argocd
```

Comment:
this will break, in self-hosted runner, there is no "current" context to reference, that is why old code creates a new context